programs/utilities/lifetimes.sc

```diff
@@ -1,74 +1,75 @@
+    ));
+	null
 );
 
-//Globals
-
-global_start=false;
-
-global_ticks=0;
-
-global_minutes=0.0;//floating point number
-
-global_mobs=m();//To keep track of which mob types are actually there
-
-global_lifetimes=m();//SO that I can measure multiple lifetimes at the same time, I have to have an entry for each mob which he puts into global_lifetimes when he dies
-
-global_lifetimes_global=m();
-
-
-//Events
+global_start_time = 0;
+
+global_lifetimes = {};
+
+global_lifetimes_categories = {};
+
+_track(e, r)->(
+	entry = global_lifetimes:str(e);
+	if(entry==null,
+		global_lifetimes:str(e) = [e~'age', 1],
+		global_lifetimes:str(e) = entry + [e~'age', 1] //can't use += cos of bug
+	);
+	
+	category = e~'category';
+	entry = global_lifetimes_categories:str(category);
+	if(entry==null,
+		global_lifetimes_categories:str(category) = [e~'age', 1],
+		global_lifetimes_categories:str(category) = entry + [e~'age', 1] //can't use += cos of bug
+	)
+);
 
-__on_tick()->(
-    if(!global_start,return());
-    global_ticks+=1;
-    global_minutes=round(global_ticks/12)/100;//it would be round(global_ticks/1200(mins)*100)/100, but I simplified
-    e=filter(entity_selector('@e'),query(_,'category')!='misc');
-    for(e,
-        if(!query(_,'has_tag','lifetime'),
-            modify(_,'tag','lifetime');
-            global_mobs:query(_,'type')+=1;
-            entity_event(_,'on_death',_(e,c)->put(global_lifetimes_global,query(e,'type'),get(global_lifetimes_global,query(e,'type'))+global_lifetimes:e));//tried to define func but nulpointerexception bs
-        );
-        global_lifetimes:_+=1
-    )
+start()->(
+	global_start_time = tick_time();
+	entity_load_handler('!misc',_(e, new)->if(new, entity_event(e, 'on_death', '_track')));
+	print('Began tracking mob lifetimes');
+	null
 );
 
-//Commands
+stop()->_reset();
 
-start()->global_start=true;
+report()->_report(false);
 
-stop()->_reset();
+report_verbose()->_report(true);
 
-report()->(
-    total_avg=0.0;
-    print('Average lifetimes for mobs (in ticks) over '+global_minutes+' minutes:');
-    for(global_mobs,
-        mob_avg=round(global_lifetimes_global:_/(global_mobs:_*6))/10
+_report(verbose)->(
+	if(global_start_time==0, exit('Nothing to report'));
+    total_time=0;
+	total_mobs=0;
+	minutes = round((tick_time()-global_start_time)/12)/100; //2 d.p
+    print('Average lifetimes for mobs (in ticks) over '+minutes+' minutes:');
+	map_to_iterate_over = if(verbose, global_lifetimes, global_lifetimes_categories);
+    for(map_to_iterate_over,
+		entry = map_to_iterate_over:_;
+        mob_avg=round(entry:0/entry:1*5)/100; //2 d.p
         print('    '+_+': '+mob_avg+' seconds');
-        total_avg=(total_avg+mob_avg)/2
+        total_time+=entry:0;
+		total_mobs+=entry:1;
     );
-    print('Total average lifetime for all mobs is: '+total_avg+' seconds')
+    print('Total average lifetime for all mobs is: '+round(total_time/total_mobs*100)/100+' seconds');
```

Comment:
this one is in ticks but says secons? I'm not sure.

Reply:
ah oops my bad, forgot to convert